librz/core/cmd_print.c

```diff
@@ -3207,59 +3207,46 @@ static void disasm_strings(RzCore *core, const char *input, RzAnalysisFunction *
 	rz_config_set_i(core->config, "emu.str", emu_str);
 }
 
-static bool cmd_print_ph(RzCore *core, const char *input) {
-	char algo[128];
-	ut32 osize = 0, len = core->blocksize;
-	const char *ptr;
-	bool handled_cmd = false;
+RZ_IPI RzCmdStatus rz_cmd_print_msg_digest_handler(RzCore *core, int argc, const char **argv) {
 	const RzMsgDigestPlugin *plugin = NULL;
-	RzCmdStateOutput state = { 0 };
-	if (!*input || *input == '?') {
-		state.mode = RZ_OUTPUT_MODE_QUIET;
-	}
-	if (*input == '=') {
-		state.mode = RZ_OUTPUT_MODE_STANDARD;
-	}
-	if (state.mode) {
-		return rz_core_hash_plugins_print(&state) ? true : false;
-	}
-	input = rz_str_trim_head_ro(input);
-	ptr = strchr(input, ' ');
-	sscanf(input, "%31s", algo);
-	if (ptr && ptr[1]) { // && rz_num_is_valid_input (core->num, ptr + 1)) {
-		int nlen = rz_num_math(core->num, ptr + 1);
-		if (nlen > 0) {
-			len = nlen;
-		}
-		osize = core->blocksize;
-		if (nlen > core->blocksize) {
-			rz_core_block_size(core, nlen);
-			if (nlen != core->blocksize) {
-				eprintf("Invalid block size\n");
-				rz_core_block_size(core, osize);
-				return false;
-			}
-			rz_core_block_read(core);
-		}
-	} else if (!ptr || !*(ptr + 1)) {
-		osize = len;
-	}
-	/* TODO: Simplify this spaguetti monster */
-	for (size_t j = 0; osize > 0 && (plugin = rz_msg_digest_plugin_by_index(j)); ++j) {
-		if (!rz_str_ccmp(plugin->name, input, ' ')) {
-			if (!strncmp(plugin->name, "entropy", 7)) {
-				handle_entropy(plugin->name, core->block, len);
-			} else {
-				handle_msg_digest(plugin->name, core->block, len);
-			}
-			handled_cmd = true;
-			break;
+
+	ut32 old_blocksize = core->blocksize;
+	ut32 new_blocksize = old_blocksize;
+
+	if (RZ_STR_ISNOTEMPTY(argv[1])) {
+		plugin = rz_msg_digest_plugin_by_name(argv[1]);
+	}
+
+	if (!plugin) {
+		RZ_LOG_ERROR("algorithm '%s' does not exists\n", argv[1]);
+		return RZ_CMD_STATUS_INVALID;
```

Comment:
```suggestion
		return RZ_CMD_STATUS_ERROR;
```
I think this is better. _INVALID is used mainly for shell/parser/grammar errors.